core/src/transport/boxed.rs

```diff
@@ -58,8 +58,11 @@ trait Abstract<O> {
         addr: Multiaddr,
     ) -> Result<(), TransportError<io::Error>>;
     fn remove_listener(&mut self, id: ListenerId) -> bool;
-    fn dial(&mut self, addr: Multiaddr) -> Result<Dial<O>, TransportError<io::Error>>;
-    fn dial_as_listener(&mut self, addr: Multiaddr) -> Result<Dial<O>, TransportError<io::Error>>;
+    fn dial(
+        &mut self,
+        addr: Multiaddr,
+        dial_opts: DialOpts,
```

Comment:
```suggestion
        opts: DialOpts,
```

Will the formatter put it on one line then? :D